app/controllers/concerns/blacklight/catalog.rb

```diff
@@ -65,7 +65,7 @@ def track
       path = uri.query ? "#{uri.path}?#{uri.query}" : uri.path
       redirect_to path, status: 303
     else
-      redirect_to blacklight_config.document_model.new(id: params[:id]), status: 303
+      redirect_to({ action: :show, id: params[:id] }, status: 303)
```

Comment:
It's not clear to me that the code you removed is not necessary. Can you explain?

Reply:
It's not necessary, because the tests show the intended behavior is unchanged.